src/content/docs/blog/2025/10/31/index.mdx

```diff
@@ -0,0 +1,155 @@
+
+### Parallelize the `moss state verify` command
+
+Until recently, when employing the `moss state verify` command to ensure the integrity of all moss states, boulder was limited to running on a single thread. The team updated the `moss state verify` command to run certain tasks in parallel (using `rayon`) which has signifantly reduced the time to complete the verification process.
+
+The `moss state verify` command currently does not have any indication of progress such as a progress bar. As such, a user may be unsure if their system is frozen. Whilst parallelizing performance-sensitive aspects of this task significantly reduces the time to completion, we have an issue raised to add a progress bar to the command for an improved user experience.
+
+### Infrastructure Stabilization
+
+As it turned out, the issue we kept seeing (and have kept seeing since May) related to builds hanging for no apparent reason, was actually related to the boulder threading code leaving threads running at the point where we called the `clone2` syscall to enter a user-namespace container for build isolation purposes -- fearless concurrency indeed...
+
+However, as these issues only manifested for us when boulder was invoked by the build infrastructure, and even then only rarely, it was not a problem we could trigger on demand.
+
+After some head-scratching and debugging sessions where we attached debuggers to live builds that had happened to hang, we finally found the root cause, and in turn were able to simplify our threading code significantly by always using a dedicated, explicit runtime that gets automatically shut down once the parallel (rayon) or threaded (tokio) work unit goes out of scope.
+
+This in turn now ensures that all threads have been shut down as required, before entering the cloned user-namespace container.
+
+Massive thanks to Cory Forsstrom for the work he put into solving this problem very elegantly over a couple of Pull Requests.
+
+### moss `system-model` prototype feature
+
+The system-model feature is inspired by the feature of the same name from the Conary system and package manager. However, in practice, it will likely end up more like a hybrid of the Conary feature and e.g. the simpler and more limited Gentoo (and FreeBSD) 'world' definition.
+
+We are still hoping to be able to, when the time comes, offer versioned system-models that match our planned versioned repository work.
+
+The goal of the moss system-model feature is that it will eventually enable us to define, and switch between, system installs declaratively and seamlessly. This feature is slated to be suitable for use on both on live systems, recovery initramfs environments, and for installing new systems.
```

Comment:
```suggestion
The goal of the moss system-model feature is that it will eventually enable us to define, and switch between, system installs declaratively and seamlessly. This feature is slated to be suitable for use on both live systems, recovery initramfs environments, and for installing new systems.
```